hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/utils/OzoneManagerRatisUtils.java

```diff
@@ -267,10 +302,13 @@ private static OMClientRequest getOMAclRequest(OMRequest omRequest) {
       } else if (ObjectType.BUCKET == type) {
         return new OMBucketAddAclRequest(omRequest);
       } else if (ObjectType.KEY == type) {
-        if (isBucketFSOptimized()){
+        OMKeyAddAclRequest aclReq = new OMKeyAddAclRequest(omRequest);
+        OmBucketInfo buckInfo = aclReq.getBucketInfo(ozoneManager);
```

Comment:
@aryangupta1998  Like we discussed for other client KeyRequests, `bucketInfo` can be null. [Reference comment](https://github.com/apache/ozone/pull/2533/files#r692388618)

Instead of returning BucketInfo, you can return BucketLayout.
Just rename method `OMKeyAclRequest.getBucketInfo` to `OMKeyAclRequestgetBucketLayout` and then return Layout itself. Here, you can handle bucketInfo null check and return BucketLayout.LEGACY.

For debugging purpose, please add warn log message saying that bucket doesn't exists.
LOG.error("Bucket not found: {}/{} ", volumeName, bucketName);

Reply:
Done